app/services/analytics_events.rb

```diff
@@ -0,0 +1,78 @@
+module AnalyticsEvents
+  # @identity.idp.event_name Account Reset
+  # @param [Boolean] success
+  # @param ["cancel", "delete", "cancel token validation", "granted token validation",
+  #  "notifications"] event
+  # @param [String] message_id from AWS Pinpoint API
+  # @param [String] request_id from AWS Pinpoint API
+  # @param [Boolean] sms_phone
+  # @param [Boolean] totp does the user have an authentication app as a 2FA option?
+  # @param [Boolean] piv_cac does the user have PIV/CAC as a 2FA option?
+  # @param [Integer] count number of notifications sent
+  # @param [Hash] errors
+  # @param [Hash] error_details
+  # @param [String] user_id
+  # @param [Integer] account_age_in_days
+  # @param [Hash] mfa_method_counts
+  # @param [Integer] email_addresses number of email addresses the user has
+  # Tracks events related to a user requesting to delete their account during the sign in process
+  # (because they have no other means to sign in).
+  def account_reset(
+    success: nil,
+    event: nil,
+    message_id: nil,
+    piv_cac: nil,
+    request_id: nil,
+    sms_phone: nil,
+    totp: nil,
+    count: nil,
+    errors: nil,
+    user_id: nil,
+    account_age_in_days: nil,
+    mfa_method_counts: nil,
+    pii_like_keypaths: nil,
+    error_details: nil,
+    email_addresses: nil
+  )
+    track_event(
+      'Account Reset',
+      {
+        success: success,
+        event: event,
+        message_id: message_id,
+        piv_cac: piv_cac,
+        request_id: request_id,
+        sms_phone: sms_phone,
+        totp: totp,
+        count: count,
+        errors: errors,
+        user_id: user_id,
+        account_age_in_days: account_age_in_days,
+        mfa_method_counts: mfa_method_counts,
+        pii_like_keypaths: pii_like_keypaths,
+        error_details: error_details,
+        email_addresses: email_addresses,
+      }.compact,
+    )
+  end
+
+  # @identity.idp.event_name Account Delete submitted
+  # @param [Boolean] success
+  # When a user submits a form to delete their account
+  def account_delete_submitted(success:)
+    track_event('Account Delete submitted', success: success)
+  end
+
+  # @identity.idp.event_name Account Delete visited
+  # When a user visits the page to delete their account
+  def account_delete_visited
```

Comment:
Turns out there are a lot of these events with no params, which would make for a super easy migration if we wanted

```
git grep -E "analytics.track_event\(Analytics::[A-Z_]+\)"
```

<details>
<summary>
Full list of events without params
<code>git grep -E "analytics.track_event\(Analytics::[A-Z_]+\)" | cut -d'(' -f 2 | cut -d')' -f 1 | sort | uniq</code>
</summary>
<pre><code>Analytics::ACCOUNT_VISIT
Analytics::AUTHENTICATION_CONFIRMATION
Analytics::AUTHENTICATION_CONFIRMATION_CONTINUE
Analytics::AUTHENTICATION_CONFIRMATION_RESET
Analytics::BACKUP_CODE_CREATED
Analytics::BANNED_USER_REDIRECT
Analytics::BANNED_USER_VISITED
Analytics::BROKEN_PERSONAL_KEY_REGENERATED
Analytics::EMAIL_LANGUAGE_VISITED
Analytics::EVENTS_VISIT
Analytics::FORGET_ALL_BROWSERS_SUBMITTED
Analytics::FORGET_ALL_BROWSERS_VISITED
Analytics::IDV_ADDRESS_VISIT
Analytics::IDV_COME_BACK_LATER_VISIT
Analytics::IDV_FORGOT_PASSWORD
Analytics::IDV_FORGOT_PASSWORD_CONFIRMED
Analytics::IDV_GPO_ADDRESS_LETTER_REQUESTED
Analytics::IDV_GPO_VERIFICATION_VISITED
Analytics::IDV_INTRO_VISIT
Analytics::IDV_PERSONAL_KEY_SUBMITTED
Analytics::IDV_PERSONAL_KEY_VISITED
Analytics::IDV_PHONE_CONFIRMATION_OTP_RATE_LIMIT_ATTEMPTS
Analytics::IDV_PHONE_CONFIRMATION_OTP_RATE_LIMIT_LOCKED_OUT
Analytics::IDV_PHONE_CONFIRMATION_OTP_RATE_LIMIT_SENDS
Analytics::IDV_PHONE_CONFIRMATION_OTP_VISIT
Analytics::IDV_PHONE_OTP_DELIVERY_SELECTION_VISIT
Analytics::IDV_PHONE_RECORD_VISIT
Analytics::IDV_REVIEW_COMPLETE
Analytics::IDV_REVIEW_VISIT
Analytics::MULTI_FACTOR_AUTH_ENTER_TOTP_VISIT
Analytics::MULTI_FACTOR_AUTH_MAX_ATTEMPTS
Analytics::MULTI_FACTOR_AUTH_MAX_SENDS
Analytics::MULTI_FACTOR_AUTH_OPTION_LIST_VISIT
Analytics::PASSWORD_MAX_ATTEMPTS
Analytics::PASSWORD_RESET_VISIT
Analytics::PENDING_ACCOUNT_RESET_CANCELLED
Analytics::PENDING_ACCOUNT_RESET_VISITED
Analytics::PERSONAL_KEY_REACTIVATION
Analytics::PERSONAL_KEY_REACTIVATION_SIGN_IN
Analytics::PERSONAL_KEY_REACTIVATION_VISITED
Analytics::PHONE_CHANGE_VIEWED
Analytics::PROFILE_PERSONAL_KEY_CREATE
Analytics::PROFILE_PERSONAL_KEY_VISIT
Analytics::PROOFING_ADDRESS_RESULT_MISSING
Analytics::PROOFING_DOCUMENT_RESULT_MISSING
Analytics::PROOFING_RESOLUTION_RESULT_MISSING
Analytics::RULES_OF_USE_VISIT
Analytics::SESSION_KEPT_ALIVE
Analytics::SESSION_TIMED_OUT
Analytics::SESSION_TOTAL_DURATION_TIMEOUT
Analytics::SP_HANDOFF_BOUNCED_DETECTED
Analytics::SP_HANDOFF_BOUNCED_VISIT
Analytics::SP_INACTIVE_VISIT
Analytics::TOTP_USER_DISABLED
Analytics::USER_REGISTRATION_ENTER_EMAIL_VISIT
Analytics::USER_REGISTRATION_PHONE_SETUP_VISIT
Analytics::USER_REGISTRATION_PIV_CAC_DISABLED
Analytics::USER_REGISTRATION_PIV_CAC_SETUP_VISIT</code></pre>
</details>